compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1112,8 +1112,10 @@ object Types {
      */
     def matches(that: Type)(using Context): Boolean = {
       record("matches")
-      withoutMode(Mode.SafeNulls)(
-        TypeComparer.matchesType(this, that, relaxed = !ctx.phase.erasedTypes))
+      val overrideCtx = if ctx.explicitNulls
+        then ctx.retractMode(Mode.SafeNulls).addMode(Mode.RelaxedOverriding)
```

Comment:
There are three occurrences of `ctx.retractMode(Mode.SafeNulls).addMode(Mode.RelaxedOverriding)`, which each create a useless intermediate context. I think it's better to define something like this:
```scala
def relaxedOverrideContext(using Context) = 
  ctx.withModeBits(ctx.mode &~ Mode.SafeNulls | Mode.RelaxedOverriding)
```

Reply:
`relaxedOverrideContext` is added to Contexts